### PR TITLE
fixed multiple realms selection

### PR DESCRIPTION
**Pull request by Yehonal**
and changed OS to pass warden check

### CODE REVIEW

File: src/lib/config.js [hunk 1]
```diff
@@ -30,7 +30,7 @@ class Config {
     this.timezone = 0;
 
     this.locale = 'enUS';
-    this.os = 'Mac';
+    this.os = 'Win';
```
Comment (fallenoak) [inline src/lib/config.js]: Can you point to where this Warden check is implemented in either TrinityCore or cmangos?

Comment (Yehonal) [inline src/lib/config.js]: https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/Server/WorldSession.cpp

In function
void WorldSession::InitWarden

However it's a check done in various part of code and trinitycore uses Win
and OSX as identifiers.

So Win is quite universal and should be used.

I've tested and it works

Sorry i'm from mobile now i can t link you the exact line
Il 23/Mar/2016 15:02, "fallenoak" <notifications@github.com> ha scritto:

Comment (fallenoak) [inline src/lib/config.js]: I see the following code, but I'm not sure I'm following why Warden fails if the identifier isn't one of 'Win' or 'OSX' -- to me, it looks like it just skips init'ing Warden altogether.

```
void WorldSession::InitWarden(BigNumber* k, std::string const& os)
{
    if (os == "Win")
    {
        _warden = new WardenWin();
        _warden->Init(this, k);
    }
    else if (os == "OSX")
    {
        // Disabled as it is causing the client to crash
        // _warden = new WardenMac();
        // _warden->Init(this, k);
    }
}
```

Comment (Yehonal) [inline src/lib/config.js]: As i said you TC does multiple checks in core and if platform is not Win or OSX it doesn't allow you connecting to worldserver.

However if you don't believe me i've found the line for you: https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/Server/WorldSocket.cpp#L471

Comment (fallenoak) [inline src/lib/config.js]: Right. Maybe this would be better as a separate commit, then? It doesn't seem related to fixing multiple realm handling, which is what the rest of your commit deals with.

Comment (Yehonal) [inline src/lib/config.js]: i've to do multiple PR or just separate the commit and include in same PR?

Comment (fallenoak) [inline src/lib/config.js]: I think separating the commit, but including it in the same PR would be great. I'm mostly thinking about what happens to tools like git blame when commits do too many separate things. :) @timkurvers What do you think?

File: src/components/auth/index.jsx [hunk 2]
```diff
@@ -34,6 +34,7 @@ class AuthScreen extends React.Component {
   }
 
   connect(host, port) {
+    session.authstate=this.state;
```
Comment (fallenoak) [inline src/components/auth/index.jsx]: Minor nitpick on code style: separate ```=``` with spaces (stays consistent with our existing code).

File: src/lib/game/handler.js [hunk 3]
```diff
@@ -27,9 +27,10 @@ class GameHandler extends Socket {
   }
 
   // Connects to given host through given port
-  connect(host, port) {
+  connect(host, realm) {
+    this.realm=realm;
```
Comment (fallenoak) [inline src/lib/game/handler.js]: Minor nitpick on code style: separate ```=``` with spaces (stays consistent with our existing code).

**Comment by timkurvers** [comment]
Let's merge #120 first, as that contains the fix for allowing multiple realms to be handled correctly. It seems to be included in here, but it's difficult to see what's new.

After that we can have a look at the Win vs OSX identifier, as well as Trinity requiring a realm ID to be passed in.

**Comment by Yehonal** [comment]
i've splitted commit and changed that minor code-style

**Comment by Yehonal** [comment]
could you accept this PR? it is fully working

**Comment by timkurvers** [comment]
#120 was merged, which contained the original fix for supporting multiple realms.

Could you rebase this on top of current master only including the TrinityCore fixes?

**Comment by Yehonal** [comment]
Splitted in: 

https://github.com/wowserhq/wowser/pull/165

https://github.com/wowserhq/wowser/pull/166

**Comment by timkurvers** [comment]
Thanks 👍
